lib/vernier/output/firefox.rb

```diff
@@ -165,6 +165,15 @@ def initialize(ruby_thread_id, profile, categorizer, name:, tid:, samples:, weig
           @tid = tid
           @name = name
 
+          if @name == ""
+            tr = ObjectSpace._id2ref(@ruby_thread_id)
```

Comment:
I think there's _probably_ a better way to do this from within the C extension so that we can capture thread names before they GC, but this seems like an okay fallback.

One problem though is `_id2ref` will raise if the thread is GC'd (they _should_ be collectible when stopped even with Vernier running). Could we make sure that is handled properly?

Reply:
> One problem though is _id2ref will raise if the thread is GC'd (they should be collectible when stopped even with Vernier running). Could we make sure that is handled properly?

Added a rescue for [RangeError](https://github.com/ruby/ruby/blob/master/gc.c#L4477-L4478) to cover this case

> Yeah, we could just record rb_inspect(thread), I'm unfamiliar with Vernier internals though, so not sure where it would be best to call this.

I decided to take a crack at this, basically set the thread name any time that the `RESUME` callback runs (since we should have the GVL), so the name will be whatever the name was when the thread was last resumed.

If we never got the inspect output in the native code, it does a best-effort attempt to grab it from ruby land.

Either way, whatever we got from inspect either in native or ruby call is then parsed to a prettier value in ruby during the output generation.